tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc

```diff
@@ -2120,11 +2120,34 @@ Status ConvertConv2DHelper(OpConverterParams* params, int group,
         {{"input_sizes", true}, {"filter", true}, {"out_backprop", false}}));
     backprop_output_size = inputs.at(0);
     tensor = inputs.at(2).tensor();
-    if (!HasStaticShape(tensor->getDimensions())) {
-      // TODO(tfeher): Allow dynamic input. We need to implement padding
-      // correction for dynamic shapes in this case.
+
+    bool has_dynamic_hw_shape{false};
+    {
+      int start_idx{0};
+      auto dims = tensor->getDimensions();
+      if (params->use_implicit_batch) {
+        if (dims.nbDims != 3) {
+          return errors::Internal(
+              "In implicit batch mode, input nbDims should be 3");
+        }
+        start_idx = 1;
+      } else {
+        if (dims.nbDims != 4) {
+          return errors::Internal(
+              "In explicit batch mode, input nbDims should be 4");
+        }
+        start_idx = 0;
```

Comment:
do you actually want "start_idx = 2" instead?

Reply:
Sorry I fat fingered a couple things. I just pushed an update